hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/FilePerChunkStrategy.java

```diff
@@ -218,17 +244,36 @@ public ChunkBuffer readChunk(Container container, BlockID blockID,
     long len = info.getLen();
     ByteBuffer data = ByteBuffer.allocate((int) len);
 
+    int index = extractChunkIndex(info.getChunkName());
+    if (index == -1) {
+      throw new StorageContainerException(
+          "Chunk file name can't be parsed " + possibleFiles.toString(),
+          UNABLE_TO_FIND_CHUNK);
+    }
+    // Chunk index start from 1
+    Preconditions.checkState(index > 0);
+
+    long chunkFileOffset;
+    try {
+      BlockData blockData = blockManager.getBlock(kvContainer, blockID);
+      List<ContainerProtos.ChunkInfo> chunks = blockData.getChunks();
+      Preconditions.checkState(index <= chunks.size());
+      chunkFileOffset = chunks.get(index - 1).getOffset();
```

Comment:
Instead of parsing the chunk name for an index, we could find the chunk with the exact same name.  I think it's less fragile, although might be slower if the block has too many chunks.  What's your opinion?

https://github.com/adoroszlai/hadoop-ozone/commit/398ea1d5991c44baa59c120dea8fa3c80727e9e2 (note: CI is still in progress, may need tweak based on the result)

Reply:
A good point. I think the performance overhead can be ignored.